File: Tests/CustomerDataTests.cs
using SomeBasicFileStoreApp.Core;
using System.Linq;
using Xunit;

namespace SomeBasicFileStoreApp.Tests
{
    public class CustomerDataTests
    {
        private static IRepository _repository;
        private static ObjectContainer _container;
        [Fact]
        public void CanGetCustomerById()
        {
            Assert.NotNull(_repository.GetCustomer(1));
        }

        [Fact]
        public void CanGetProductById()
        {
            Assert.NotNull(_repository.GetProduct(1));
        }
        [Fact]
        public void OrderContainsProduct()
        {
            Assert.Contains(_repository.GetOrder(1).Products, p => p.Id == 1);
        }
        [Fact]
        public void OrderHasACustomer()
        {
            Assert.False(string.IsNullOrWhiteSpace(_repository.GetOrder(1).Customer.Name.First));
        }
        static CustomerDataTests()
        {
            _container = new ObjectContainer();
            //_container.Boot();
            _repository = _container.GetRepository();
            var commands = new GetCommands().Get();
            _container.HandleAll(commands);
        }
    }
}

File: CoreFsTests/CustomerDataTests.fs
namespace CoreFsTests

open Xunit
open System.IO
open System
open System.Collections.Generic

open SomeBasicFileStoreApp
open Helpers
open GetCommands

type CustomerDataTests()=
    let _container = new ObjectContainer()
    let _repository = _container.GetRepository();

    do
        //_container.Boot();
        let commands = getCommands()
        _container.Handle(commands |> unwrap)

    [<Fact>]
    member this.CanGetCustomerById()=
        Assert.NotNull(_repository.GetCustomer(1))

    [<Fact>]
    member this.CanGetProductById()=
        Assert.NotNull(_repository.GetProduct(1))

    [<Fact>]
    member this.OrderContainsProduct()=
        let order = _repository.GetOrder(1)
        Assert.True(order.Products |> List.tryFind( fun p -> p.Id = 1) |> Option.isSome)

    //[<Test>]
    //member this.OrderHasACustomer()=
    //    Assert.IsNotNullOrEmpty(_repository.GetTheCustomerOrder(1).Firstname)